sdk/storage/Azure.Storage.Common/src/Shared/StorageWriteStream.cs

```diff
@@ -196,5 +197,22 @@ protected static void ValidateWriteParameters(byte[] buffer, int offset, int cou
                 throw new ArgumentOutOfRangeException($"{nameof(offset)} + {nameof(count)} cannot exceed {nameof(buffer)} length.");
             }
         }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                Flush();
+            }
+
+            disposed = true;
```

Comment:
How does `PooledMemoryStream _buffer` get cleaned up?  Should we dispose it if we own the lifetime because we created it in the .ctor?  We want to make sure we don't exhaust the ArrayPool by not returning memory to it.

Reply:
Good catch. Added handling for that case. Otherwise I guess it would wait for GC.